### PR TITLE
Fixed course view base classes

### PR DESCRIPTION
**Pull request by clintonb**
Paying some code debt and cleaning the inheritance. Functionality should remain unchanged.

@rocha @dsjen @mulby @jhelbert @brianhw

### CODE REVIEW

File: analytics_dashboard/courses/views.py [hunk 1]
```diff
@@ -4,46 +4,31 @@
...
+    Adds default course context data.
 
-    # displayed on the page
+    Use primarily with templated views where data needs to be passed to Javascript.
+    """
+    # Title displayed on the page
     page_title = None
-    # page returned for tracking
+
+    # Page name used for event tracking
```
Comment (rocha) [inline analytics_dashboard/courses/views.py]: Minor: We use the term "event tracking" in other places with a different meaning. Maybe replace with "product analytics" or "usage analytics"?

Comment (clintonb) [inline analytics_dashboard/courses/views.py]: Done.

File: analytics_dashboard/courses/views.py [hunk 2]
```diff
@@ -81,12 +66,23 @@ def get_default_data(self):
...
     course = None
+    course_id = None
 
-    def render_to_response(self, context, **response_kwargs):
-        raise NotImplementedError
+    def dispatch(self, request, *args, **kwargs):
+        self.course_id = kwargs['course_id']
+
+        try:
+            return super(CourseView, self).dispatch(request, *args, **kwargs)
```
Comment (rocha) [inline analytics_dashboard/courses/views.py:83]: I am not sure how this works. Let me see if I understand: here we basically call dispatch of the parent class (`TemplateView`) and that in turn will call a method on a derive class, correct?

Comment (clintonb) [inline analytics_dashboard/courses/views.py:83]: Correct. `dispatch()` will most likely (depending on the request type) call `get()` which will do the rendering. The assumption is that the downstream rendering uses the API client which will raise a `NotFoundError` if a course is not found.

Comment (rocha) [inline analytics_dashboard/courses/views.py:83]: Ok, makes sense. However, is there a better way to pass `course_id` downstream. If a subclass of `CourseView` overrides `dispatch` and doesn't call `super`... nah that seems to be a bad usage case.

**Comment by rocha** [comment]
:rocket:

File: analytics_dashboard/courses/views.py (hunk 2 above)
Comment (clintonb) [inline analytics_dashboard/courses/views.py:83]: I thought about that. The subclass implementor needs to make sure he/she doesn't do that.

**Pull request merged**